fe/src/main/java/org/apache/doris/external/elasticsearch/EsRestClient.java

```diff
@@ -17,92 +17,183 @@
         Map<String, Map<String, Object>> nodesData = get("_nodes/http", "nodes");
         if (nodesData == null) {
             return Collections.emptyMap();
         }
-        Map<String, EsNodeInfo> nodes = new HashMap<>();
+        Map<String, EsNodeInfo> nodesMap = new HashMap<>();
         for (Map.Entry<String, Map<String, Object>> entry : nodesData.entrySet()) {
             EsNodeInfo node = new EsNodeInfo(entry.getKey(), entry.getValue());
             if (node.hasHttp()) {
-                nodes.put(node.getId(), node);
+                nodesMap.put(node.getId(), node);
             }
         }
-        return nodes;
+        return nodesMap;
     }
-
-    public String getIndexMetaData(String indexName) throws Exception {
-        String path = "_cluster/state?indices=" + indexName
-                + "&metric=routing_table,nodes,metadata&expand_wildcards=open";
-        return execute(path);
-
+    
+    public EsFieldInfo getFieldInfo(String indexName, String mappingType, List<Column> colList) throws Exception {
+        String path = indexName + "/_mapping";
+        String indexMapping = execute(path);
+        if (indexMapping == null) {
+            throw new Exception( "index[" + indexName + "] _mapping not found for the Elasticsearch Cluster");
+        }
+        return getFieldInfo(colList, parseProperties(indexMapping, mappingType));
     }
-
+    
+    @VisibleForTesting
+    public EsFieldInfo getFieldInfo(List<Column> colList, JSONObject properties) {
+        if (properties == null) {
+            return null;
+        }
+        Map<String, String> fetchFieldMap = new HashMap<>();
+        Map<String, String> docValueFieldMap = new HashMap<>();
+        for (Column col : colList) {
+            String colName = col.getName();
+            if (!properties.has(colName)) {
+                continue;
+            }
+            JSONObject fieldObject = properties.optJSONObject(colName);
+            String fetchField = EsUtil.getFetchField(fieldObject, colName);
+            if (StringUtils.isNotEmpty(fetchField)) {
+                fetchFieldMap.put(colName, fetchField);
+            }
+            String docValueField = EsUtil.getDocValueField(fieldObject, colName);
+            if (StringUtils.isNotEmpty(docValueField)) {
+                docValueFieldMap.put(colName, docValueField);
+            }
+        }
+        return new EsFieldInfo(fetchFieldMap, docValueFieldMap);
+    }
+    
+    @VisibleForTesting
+    public JSONObject parseProperties(String indexMapping, String mappingType) {
+        JSONObject jsonObject = new JSONObject(indexMapping);
+        // the indexName use alias takes the first mapping
+        Iterator<String> keys = jsonObject.keys();
+        String docKey = keys.next();
+        JSONObject docData = jsonObject.optJSONObject(docKey);
+        JSONObject mappings = docData.optJSONObject("mappings");
+        JSONObject rootSchema = mappings.optJSONObject(mappingType);
```

Comment:
rootChema maybe null, you should process this

Reply:
ok，I forgot about es7

Reply:
emm, if user set the wrong type the index does not exists, this would throw NullPointerException.